pure.zsh

```diff
@@ -517,7 +517,9 @@ prompt_pure_state_setup() {
 		who_out=$(who -m 2>/dev/null)
 		if (( $? )); then
 			# Who am I not supported, fallback to plain who.
-			who_out=$(who 2>/dev/null | grep ${TTY#/dev/})
+			declare -a who_in
+			who_in=( ${(f)"$(who 2>/dev/null)"} )
+			who_out="${(M)who_in:#*${TTY#/dev/}*}"
```

Comment:
I noticed a bug in the previous implementation which we could solve by changing the matching ever so slightly => `*${TTY#/dev/}[^0-9]*`. I did a quick test and this seems to work, can you find any problems with it or a better approach?

So the problem is that `pts/1` matches `pts/12` (as an example), in both this PR and previous implementation, this would produce multiple matches which is not what we want.

Reply:
`*[:space:]${TTY#/dev/}[:space:]*` would be safe to the point of paranoid, which is always good.